service_config.md

```diff
@@ -0,0 +1,194 @@
+                     |                              | [service_user]                                                                     |
+                     |                              | password = {{.Password}}                                                           |
+                     |                              +------------------------------------------------------------------------------------+
+                     |
++----------------------------------------+      +----------------------------------------------------------------------------------------+
+| apiVersion: v1                   (2)   |      |  “config_files”: [        (3)                                                          |
+| metadata:                              |      |                  ...                                                                   |
+|   name: 01-<service>-deployment-secret |      |      {                                                                                 |
+|   namespace: openstack                 | ===> |        "source": "/var/lib/config-data/deployment/01-deployment-<service>-secret.conf",|
+| stringData:                            |      |        "dest": "/etc/<service>/<service>.conf.d/01-deployment-<service>-secret.conf",  |
+|   01-deployment-<service>.conf: |      |      |        "owner": "<service>",                                                           |
+|        <data>                          |      |        "perm": "0600"                                                                  |
++----------------------------------------+      |      },                                                                                |
+                                                +----------------------------------------------------------------------------------------+
+```
+
+As the diagram above depicts, a given operator is supposed to implement the
+logic to build and reconcile the `Secret` when the template has been processed.
+The `Secret`, mounted to the resulting service Pod, in the very last step of
+this process is synced by kolla to the destination directory: instead of having
+a bash script (e.g. `init.sh`) doing the `chown` on the destination folder,
+we'll rely (where possible) on kolla that allows setting ownership and
+permissions, as well as copying optional files in case additional configuration
+is provided.
+
+
+## Render multiple secrets: “CustomServiceConfigSecret” interface
+
+Currently the `OpenStack` storage operators expose the
+`customConfigServiceSecret` parameter, which provides a mechanism for
+specifying service configs via Secrets. Instead of specifying sensitive config
+data directly in the `customServiceConfig`, a cloud admin can place sensitive
+data in a `Secret`, and reference the secret by name in the service's
+`customServiceConfigSecrets` as a list of `Secret` names. Such parameter is
```

Comment:
This section references both `customConfigServiceSecret` and the plural version `customConfigServiceSecrets`.  Then below on L 92 it mentions the singular `customConfigServiceSecret` again.  Just trying to understand which one it should actually be, or is this use of both singular and plural intentional?

Reply:
I think it should be 'customServiceConfigSecret'

Reply:
sorry, my fault there, `customServiceConfigSecrets` (plural) is an array of secrets: they can be referenced by `name` in the CR and the result is that `04-glance-secrets.conf` that has, as a result, the concatenation of the secret Data.
```
for _, name := range instance.Spec.CustomServiceConfigSecrets {
		confSecret, _, err := oko_secret.GetSecret(ctx, helper, name, instance.Namespace)
		if err != nil {
			// Secret not found or unable to retrieve the secret, returning err
			return ctrlResult, err
		}
		confSecrets = append(confSecrets, *confSecret)
	}
```

I'll patch the plural in the next iteration.

Reply:
Done